### ISSUE TITLE
[makeotf] UnicodeDecodeError: 'ascii' codec can't decode byte

### ISSUE THREAD

**Issue by miguelsousa** [adobe-type-tools/afdko#847]
**makeotf** is unable to print a non-ASCII message coming from **makeotfexe**.

```
makeotfexe -f font.pfa -ff feat.fea 
makeotfexe [WARNING] unhinted <a>
makeotfexe [WARNING] glyph 'negative' has a negative advance width. Setting width to 0.
makeotfexe [WARNING] FontMenuNameDB file was not specified or not found. [SourceSansPro-Regular]
makeotfexe [FATAL] <SourceSansPro-Regular> invalid token (text was "?") [feat.fea 7]
```

```
makeotf -f font.pfa -ff feat.fea 
makeotf [Warning] Could not find FontMenuNameDB file. Font will be built with menu names derived from PostScript name.
makeotfexe [WARNING] glyph 'negative' has a negative advance width. Setting width to 0.
makeotfexe [WARNING] FontMenuNameDB file was not specified or not found. [SourceSansPro-Regular]
Traceback (most recent call last):
  File "makeotf", line 11, in <module>
    load_entry_point('afdko', 'console_scripts', 'makeotf')()
  File "makeotf.py", line 2720, in main
    runMakeOTF(makeOTFParams)
  File "makeotf.py", line 2596, in runMakeOTF
    fdkutils.runShellCmdLogging(commandString)
  File "fdkutils.py", line 122, in runShellCmdLogging
    print(tostr(output))
  File "fontTools/misc/py23.py", line 131, in tounicode
    return s.decode(encoding, errors)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xce in position 68: ordinal not in range(128)
```

[test_files.zip](https://github.com/adobe-type-tools/afdko/files/3395673/test_files.zip)

**Comment by josh-hadley** [comment]
A data point: with current AFDKO (WIP, removing use of `py23` and others), I get:
```bash
makeotf -f font.pfa -ff feat.fea
makeotf [Warning] Could not find FontMenuNameDB file. Font will be built with menu names derived from PostScript name.
b"makeotfexe [WARNING] glyph 'negative' has a negative advance width. Setting width to 0."
b'makeotfexe [WARNING] FontMenuNameDB file was not specified or not found. [SourceSansPro-Regular]'
b'makeotfexe [FATAL] <SourceSansPro-Regular> invalid token (text was "\xce") [feat.fea 7]'
makeotf [Error] Failed to build output font file 'SourceSansPro-Regular.otf'.
```
(and exit code of `1` from `makeotf`)

**Comment by miguelsousa** [comment]
@josh-hadley OK. You still need to have a bytes to string conversion in place to remove the `b` prefix from the messages.

**Comment by josh-hadley** [comment]
Agreed re: string conversion. Do you want to open that as a separate issue or should I just put it into my current WIP?

**Comment by miguelsousa** [comment]
Do it as part of your WIP.

**Comment by miguelsousa** [comment]
Interestingly `feaLib` has no problems making a GSUB table from the non-ASCII features file.

    fonttools feaLib feat.fea font.otf

[feaLib_test.zip](https://github.com/adobe-type-tools/afdko/files/3398393/feaLib_test.zip)

**Comment by khaledhosny** [comment]
See https://github.com/adobe-type-tools/afdko/issues/165#issuecomment-279779317, feaLib does not enforce the ASCII-only restriction (unintentionally I think, but for good anyway).

**Comment by miguelsousa** [comment]
@khaledhosny agreed. I'm all for supporting UTF-8 in feature syntax.

**Comment by josh-hadley** [comment]
Better?
```
makeotf -f font.pfa -ff feat.fea
makeotf [Warning] Could not find FontMenuNameDB file. Font will be built with menu names derived from PostScript name.
makeotfexe [WARNING] glyph 'negative' has a negative advance width. Setting width to 0.
makeotfexe [WARNING] FontMenuNameDB file was not specified or not found. [SourceSansPro-Regular]
makeotfexe [FATAL] <SourceSansPro-Regular> invalid token (text was "\xce") [feat.fea 7]
makeotf [Error] Failed to build output font file 'SourceSansPro-Regular.otf'.
```

**Comment by miguelsousa** [comment]
much better

**Comment by josh-hadley** [comment]
I updated `get_shell_command_output`, `runShellCmd`, and `runShellCmdLogging` in `fdkutils.py`, simplified to decode shell/subprocess output with `.decode('utf-8', 'backslashreplace')`. The particular issue above was in `runShellCmdLogging` but all 3 were previously doing similar things with subprocess output.

I'll close this out later (after PR for current WIP).

**Comment by josh-hadley** [comment]
Addressed in #850
